x-pack/solutions/security/plugins/elastic_assistant/server/lib/langchain/output_chunking/nodes/refine/helpers/get_combined_refine_prompt/index.ts

```diff
@@ -5,31 +5,36 @@
  * 2.0.
  */
 
-import type { DefendInsight } from '@kbn/elastic-assistant-common';
 import { isEmpty } from 'lodash/fp';
 
+import type { GraphInsightTypes } from '../../../../../graphs';
+
 /**
  * Returns a prompt that combines the initial query, a refine prompt, and partial results
  */
-export const getCombinedRefinePrompt = ({
+export const getCombinedRefinePrompt = <T extends GraphInsightTypes>({
   prompt,
   combinedRefinements,
+  continuePrompt,
   refinePrompt,
   unrefinedResults,
-  continuePrompt,
 }: {
   prompt: string;
   combinedRefinements: string;
-  refinePrompt: string;
   continuePrompt: string;
-  unrefinedResults: DefendInsight[] | null;
+  refinePrompt: string;
+  unrefinedResults: T[] | null;
 }): string => {
   const baseQuery = `${prompt}
 
 ${refinePrompt}
 
 """
-${JSON.stringify(unrefinedResults, null, 2)}
+\`\`\`json
+{
+  "insights": ${JSON.stringify(unrefinedResults, null, 2)}
+}
+\`\`\`
```

Comment:
This fixes an intermittent refinement step error, detailed in https://github.com/elastic/kibana/pull/215816 and addressed in commit https://github.com/elastic/kibana/pull/216757/commits/48ded9315a027bc5a7eed2a1eb07741cb976af98